### PR TITLE
fix: app crash when use a JSON without ScreenComponent on startActivity

### PR DESCRIPTION
**Pull request by viniciusguardieirozup**
### Related Issues

#973 

### Description and Example

Crash app when startActivity is called using a JSON without a ScreenComponent, this happen because in the method onResume on BeagleActivity, before call the fetch was doing a cast to ScreenComponent 

### Checklist

- [x] I read the [PR Guide] and followed the process outlined there for submitting this PR.
- [x] I avoided _breaking changes_ by not changing public APIs that people rely on. 
- [x] I am willing to follow-up on review comments in a timely manner.

### CODE REVIEW

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">The title of PR not follow the rules of: https://www.conventionalcommits.org/en/v1.0.0/</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: android/beagle/src/test/java/br/com/zup/beagle/android/view/viewmodel/BeagleViewModelTest.kt [hunk 1]
```diff
@@ -74,11 +81,13 @@ class BeagleViewModelTest {
...
         coEvery { actionRequester.fetchAction(any()) } returns action
         every { observer.onChanged(any()) } just Runs
+        coEvery { observer.onChanged(capture(slotViewState)) } just Runs
+
     }
 
     @Test
     @Suppress("UNCHECKED_CAST")
-    fun fetch_should_return_render_ViewState() {
+    fun `fetch should return render ViewState`() {
```
Comment (diogooliveira-zup) [inline android/beagle/src/test/java/br/com/zup/beagle/android/view/viewmodel/BeagleViewModelTest.kt:90]: Hi! Please add GIVEN WHEN THEN pattern to the test

File: android/beagle/src/test/java/br/com/zup/beagle/android/view/viewmodel/BeagleViewModelTest.kt [hunk 2]
```diff
@@ -94,7 +103,7 @@ class BeagleViewModelTest {
     }
 
     @Test
-    fun fetch_should_return_a_error_ViewState() {
+    fun `fetch should return a error ViewState`() {
```
Comment (diogooliveira-zup) [inline android/beagle/src/test/java/br/com/zup/beagle/android/view/viewmodel/BeagleViewModelTest.kt:106]: The same above

File: android/beagle/src/test/java/br/com/zup/beagle/android/view/viewmodel/BeagleViewModelTest.kt [hunk 3]
```diff
@@ -112,12 +121,11 @@ class BeagleViewModelTest {
     }
 
     @Test
-    fun fetch_should_return_a_error_ViewState_retry() {
+    fun `fetch should return a error ViewState retry`() {
```
Comment (diogooliveira-zup) [inline android/beagle/src/test/java/br/com/zup/beagle/android/view/viewmodel/BeagleViewModelTest.kt:124]: The same above

**Review by diogooliveira-zup — CHANGES_REQUESTED** [review]

**Review by diogooliveira-zup — APPROVED** [review]

**Review by uziasferreirazup — APPROVED** [review]

**Pull request merged**